dynet/CMakeLists.txt

```diff
@@ -196,11 +192,28 @@ endif()
 # Build cpu library
 add_library(dynet ${dynet_library_SRCS} ${dynet_library_HDRS})
 target_link_libraries(dynet ${LIBS})
+
 if(WITH_CUDA_BACKEND)
-  foreach(FILENAME ${dynet_gen_file_prefix_SRCS})
-    set(FILEPREFIX ${FILENAME})
-    configure_file(gpu-file-template.cu.in gpu-${FILENAME}.cu)
-  endforeach()
+  if(${GPU_NUMFILES} EQUAL 0)
```

Comment:
I guess I should have left some comments in this part of the code. When GPU_NUMFILES is 0, it generates the ~30 files as before, with specific file naming. If we didn't care about the naming, then this if statement could be simplified to just set GPU_NUMFILES to the length of the list of merge files, but I think it's nice when not doing any merging to keep the files named like "gpu-nodes-flow.cu" rather than "gpu-merge0.cu".